distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -1543,6 +1519,252 @@ public void testFederatedDownloadProductToCacheOnlyCacheDisabled() throws Except
+                NOTIFICATIONS_CHANNEL)
+                .size() + " of 1").within(10, SECONDS)
+                .until(() -> adminCometDClient.getMessages(NOTIFICATIONS_CHANNEL)
+                        .size() == 1);
+        expect("Waiting for activities. Received " + adminCometDClient.getMessages(
+                ACTIVITIES_CHANNEL)
+                .size() + " of 2").within(10, SECONDS)
+                .until(() -> adminCometDClient.getMessages(ACTIVITIES_CHANNEL)
+                        .size() == 2);
+
+        List<String> adminNotifications = adminCometDClient.getMessagesInAscOrder(
+                NOTIFICATIONS_CHANNEL);
+        assertThat(adminNotifications.size(), is(1));
+        CometDMessageValidator.verifyNotification(JsonPath.from(adminNotifications.get(0)),
+                filename,
+                String.format("Resource retrieval completed, %d bytes retrieved. ",
+                        resourceData.length()),
+                "complete");
+    }
+
+    @Test
+    public void testSingleUserDownloadSameProductAsync() throws Exception {
+        getSecurityPolicy().configureWebContextPolicy(null,
+                "/=SAML|basic,/solr=SAML|PKI|basic",
+                null,
+                null);
+
+        adminCometDClient = setupCometDClientWithUser(Arrays.asList(NOTIFICATIONS_CHANNEL,
+                ACTIVITIES_CHANNEL), "admin", "admin");
+
+        String filename = "product4.txt";
+        String metacardId = generateUniqueMetacardId();
+        String resourceData = getResourceData(metacardId);
+
+        setupStubCswResponse(filename, metacardId, resourceData);
+
+        String resourceDownloadUrlLocalhostUserAsync =
+                RESOURCE_DOWNLOAD_ENDPOINT_ROOT.getUrl() + "?source=" + CSW_STUB_SOURCE_ID
+                        + "&metacard=" + metacardId;
+
+        // Download product twice via async, should only call the stub server to download once
+        // @formatter:off
+        String downloadId = given().auth().preemptive().basic("admin", "admin").when()
+                .get(resourceDownloadUrlLocalhostUserAsync).then().log().all()
+                .extract().jsonPath().getString("downloadId");
+        assertThat(downloadId, is(notNullValue()));
+        String downloadId2 = given().auth().preemptive().basic("admin", "admin").when()
+                .get(resourceDownloadUrlLocalhostUserAsync).then().log().all()
+                .extract().jsonPath().getString("downloadId");
+        assertThat(downloadId2, is(nullValue()));
+
+        verifyCswStubCall(1, metacardId);
+
+        expect("Waiting for notifications. Received " + adminCometDClient.getMessages(
+                NOTIFICATIONS_CHANNEL)
+                .size() + " of 1").within(10, SECONDS)
+                .until(() -> adminCometDClient.getMessages(NOTIFICATIONS_CHANNEL)
+                        .size() == 1);
+        expect("Waiting for activities. Received " + adminCometDClient.getMessages(
+                ACTIVITIES_CHANNEL)
```

Comment:
We could consider takingout the expect activities since we don't assert against any of the messages.